packages/pyright-internal/src/backgroundThreadBase.ts

```diff
@@ -93,69 +92,73 @@ export class BackgroundThreadBase {
     }
 }
 
-export function createJsonObjectFrom(configOptions: ConfigOptions): any {
-    return {
-        ...JSON.parse(JSON.stringify(configOptions)),
-        defineConstant: configOptions.defineConstant, // This is the only field that can't be stringified.
-    };
+// Function used to serialize specific types that can't automatically be serialized.
+// Exposed here so it can be reused by a caller that wants to add more cases.
+export function serializeReplacer(key: string, value: any) {
```

Comment:
It looks like you're relying on these special names being unique and never appearing elsewhere in our code. That's probably an OK assumption because you're using a non-standard (Python-style) naming convention here, but it might be safer to use an even more obscure naming convention like `__serialization_mapValue` or something like that.

Reply:
Good idea. I'll change it to that.